### ISSUE TITLE
helm init with TLS getting "Error: could not ping Tiller: context deadline exceeded"

### ISSUE THREAD

**Issue by ruaridhnewman** [helm/helm#7161]
## Issue

I am trying to install helm in an AWS eks cluster, with TLS enabled. To do this, I have followed the steps outlined in [this blog post](https://medium.com/google-cloud/install-secure-helm-in-gke-254d520061f7):

 - using openssl to generate
    - `tiller.cert.pem`
    - `tiller.key.pem`
    - `ca.cert.pem`
 - run the following commands to initialise helm and tiller:

```bash
    kubectl -n kube-system create sa tiller

    kubectl create clusterrolebinding tiller-cluster-rule \
        --clusterrole=cluster-admin \
        --serviceaccount=kube-system:tiller

    helm init \
      --service-account tiller \
      --tiller-image gcr.io/kubernetes-helm/tiller:v2.14.3 \
      --tiller-tls --tiller-tls-cert ./tiller.cert.pem \
      --tiller-tls-key ./tiller.key.pem \
      --tiller-tls-verify --tls-ca-cert ca.cert.pem \
      --upgrade \
      --wait 
```

The cluster role binding and service account are both created, but on running `helm init...` the command hangs for a while then I receive the response (when running with the `--debug` flag):

```bash
[debug] Created tunnel using local port: '7381'

[debug] SERVER: "127.0.0.1:7381"

[debug] Host="", Key="./tiller.key.pem", Cert="./tiller.cert.pem", CA="ca.cert.pem"

Error: could not ping Tiller: context deadline exceeded
```

Looking inside the cluster, a `tiller-deploy` pod has been created, and logging it I get:

```bash
[main] 2019/12/05 10:53:32 Starting Tiller v2.14.3 (tls=true)
[main] 2019/12/05 10:53:32 GRPC listening on :44134
[main] 2019/12/05 10:53:32 Probes listening on :44135
[main] 2019/12/05 10:53:32 Storage driver is ConfigMap
[main] 2019/12/05 10:53:32 Max history per release is 0
```

There are also a tiller deployment and service  in the `kube-system` namespace. When trying to run any `helm` commands with the certificates, for example:

```
helm ls --tls 
    --tls-ca-cert ca.cert.pem 
    --tls-cert helm.cert.pem 
    --tls-key helm.key.pem
```

the request times out with the response `Error: context deadline exceeded`. I have also tried adding the certificate to my helm home and running `helm ls --tls`, but get the same response. With the `--debug` flag, I get this output:

```
[debug] Created tunnel using local port: '8050'

[debug] SERVER: "127.0.0.1:8050"

[debug] Host="", Key="C:\\Users\\ruaridh\\.helm\\key.pem", Cert="C:\\Users\\ruaridh\\.helm\\cert.pem", CA="C:\\Users\\ruaridh\\.helm\\ca.pem"

Error: context deadline exceeded
```

When I initialise helm without TLS on the same cluster (after cleaning the cluster), it works as expected.

---
### Other info

Output of `helm version`:
`Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}`
Server: Times out before retrieving

Output of `kubectl version`:
`Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:11:03Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-eks-b8860f", GitCommit:"b8860f6c40640897e52c143f1b9f011a503d6e46", GitTreeState:"clean", BuildDate:"2019-11-25T00:55:38Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}`

Cloud Provider/Platform: AWS

Result of `kubectl logs tiller-deploy-xxxxxxxxx-yyyyyy -n kube-system`:

```
Name:               tiller-deploy-xxxxxxxxx-yyyyyy
Namespace:          kube-system
Priority:           0
PriorityClassName:  <none>
Node:               ip-xx-x-xx-xxx.eu-west-2.compute.internal/xx.x.xx.xxx
Start Time:         Thu, 05 Dec 2019 10:53:31 +0000
Labels:             app=helm
                    name=tiller
                    pod-template-hash=xxxxxxxxx
Annotations:        kubernetes.io/psp: eks.privileged
Status:             Running
IP:                 xx.x.x.xxx
Controlled By:      ReplicaSet/tiller-deploy-xxxxxxxxx
Containers:
  tiller:
    Container ID:   docker://68690e701655b92341541b01653bb03675bdd211f72f67fb6b32ff3a32022058
    Image:          gcr.io/kubernetes-helm/tiller:v2.14.3
    Image ID:       docker-pullable://gcr.io/kubernetes-helm/tiller@sha256:c09393087c4be55023f86be979a9dfe486c728703eba996344adc9783f261baa
    Ports:          44134/TCP, 44135/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Running
      Started:      Thu, 05 Dec 2019 10:53:32 +0000
    Ready:          True
    Restart Count:  0
    Liveness:       http-get http://:44135/liveness delay=1s timeout=1s period=10s #success=1 #failure=3
    Readiness:      http-get http://:44135/readiness delay=1s timeout=1s period=10s #success=1 #failure=3
    Environment:
      TILLER_NAMESPACE:    kube-system
      TILLER_HISTORY_MAX:  0
      TILLER_TLS_VERIFY:   1
      TILLER_TLS_ENABLE:   1
      TILLER_TLS_CERTS:    /etc/certs
    Mounts:
      /etc/certs from tiller-certs (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from tiller-token-xxxxx (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  tiller-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  tiller-secret
    Optional:    false
  tiller-token-xxxxx:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  tiller-token-xxxxx
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason     Age   From                                                Message
  ----    ------     ----  ----                                                -------
  Normal  Scheduled  22m   default-scheduler                                   Successfully assigned kube-system/tiller-deploy-xxxxxxxxx-yyyyyy to ip-xx-x-xx-xxx.eu-west-2.compute.internal
  Normal  Pulled     22m   kubelet, ip-xx-x-xx-xxx.eu-west-2.compute.internal  Container image "gcr.io/kubernetes-helm/tiller:v2.14.3" already present on machine
  Normal  Created    22m   kubelet, ip-xx-x-xx-xxx.eu-west-2.compute.internal  Created container tiller
  Normal  Started    22m   kubelet, ip-xx-x-xx-xxx.eu-west-2.compute.internal  Started container tiller
```

**Comment by ruaridhnewman** [comment]
For anyone who is stuck on the same issue, I ended up following the [flux documentation](https://docs.fluxcd.io/projects/helm-operator/en/latest/tutorials/get-started.html) and using [`cfssl`](https://github.com/cloudflare/cfssl) to create the certificates.

I ran this command to initialize helm:

```
  helm init --upgrade --service-account tiller --history-max 10 \
      --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}' \
      --tiller-tls \
      --tiller-tls-cert ./tls/server.pem \
      --tiller-tls-key ./tls/server-key.pem \
      --tiller-tls-verify \
      --tls-ca-cert ./tls/ca.pem \
```

and it installed it correctly. If I ran the above command with the `--wait` flag, I had the same issue as before with it timing out. It looks like the `--wait` flag doesnt use the correct tls certificates, and I couldn't find a way to apply them manually. Instead I used the command:

`kubectl wait --for=condition=Ready pods --all -n kube-system`

to wait for the tiller pod to be ready.
